examples/now_in_dart_flutter/lib/features/detail/core/data/detail_remote_service.dart

```diff
@@ -0,0 +1,86 @@
+import 'package:dio/dio.dart';
+import 'package:fpdart/fpdart.dart';
+import 'package:meta/meta.dart';
+import 'package:now_in_dart_flutter/core/data/data.dart';
+import 'package:now_in_dart_flutter/core/domain/failure.dart';
+
+typedef _FailureOrRemoteResponse = TaskEither<Failure, RemoteResponse<String>>;
+
+abstract class DetailRemoteService {
+  DetailRemoteService({
+    Dio? dio,
+    HeaderCache? headerCache,
+  })  : _dio = dio ?? Dio(),
+        _headerCache = headerCache ?? GithubHeaderCache();
+
+  final Dio _dio;
+  final HeaderCache _headerCache;
+
+  @protected
+  @visibleForTesting
+  _FailureOrRemoteResponse getDetail(
+    int id,
+    String fullPathToMarkdownFile,
+  ) {
+    return TaskEither.Do(
+      (_) async {
+        final requestUri = await _(
+          uriParser(fullPathToMarkdownFile).toTaskEither(),
+        );
+
+        final cachedHeader = await _(
+          _headerCache.getHeader(fullPathToMarkdownFile).toTaskEither(),
+        );
+
+        return _(
+          TaskEither<Failure, Response<String>>.tryCatch(
+            () => _dio.getUri<String>(
+              requestUri,
+              options: Options(
+                headers: <String, String>{
+                  'If-None-Match': cachedHeader?.eTag ?? '',
+                },
+              ),
+            ),
+            (e, stackTrace) {
+              return ApiFailure(
+                'Error on network request',
+                errorObject: e,
+                stackTrace: stackTrace,
+              );
+            },
+          ).flatMap(
+            (response) {
+              return TaskEither<Failure, RemoteResponse<String>>(
+                () async {
+                  if (response.statusCode == 200) {
+                    final header = GithubHeader.parse(
+                      id,
+                      response,
+                      fullPathToMarkdownFile,
+                    );
+
+                    await _(_headerCache.saveHeader(header).toTaskEither());
+
+                    final html = response.data!;
+                    return right(ModifiedRemoteResponse(html));
+                  }
+
+                  return right(const UnModifiedRemoteResponse());
```

Comment:
Since you are always returning `right` this indicates that this `TaskEither` never fails, better to use `Task` then

Reply:
Something like this?
```dart
.flatMap(
  (response) {
    return Task<RemoteResponse<String>>.Do(
      (_) async {
        if (response.statusCode == 200) {
          final header = GithubHeader.parse(
            id,
            response,
            fullPathToMarkdownFile,
          );

          await _(_headerCache.saveHeader(header));

          final html = response.data!;
          return ModifiedRemoteResponse(html);
        }

        return const UnModifiedRemoteResponse();
      },
    ).toTaskEither();
  },
)
```